src/check/parse/IR.zig

```diff
@@ -1476,6 +1825,60 @@ pub const NodeStore = struct {
         };
     };
 
+    pub const TypeHeader = struct {
+        name: TokenIdx,
+        args: []const TokenIdx,
+        region: Region,
+    };
+
+    pub const TypeAnno = union(enum) {
+        ty_var: struct {
+            tok: TokenIdx,
+            region: Region,
+        },
+        star: struct {
+            region: Region,
+        },
+        underscore: struct {
+            region: Region,
+        },
+        tag: struct {
+            tok: TokenIdx,
+            args: []const TypeAnnoIdx,
+            region: Region,
+        },
+        tag_union: struct {
+            tags: []const TypeAnnoIdx,
+            open_anno: ?TypeAnnoIdx,
+            region: Region,
+        },
+        tuple: struct {
+            annos: []const TypeAnnoIdx,
+            region: Region,
+        },
+        record: struct {
+            fields: []const AnnoRecordFieldIdx,
+            region: Region,
+        },
+        @"fn": struct {
```

Comment:
Slight preference to picking a non-reserved name over needing escaping like this